lib/puppet/indirector/facts/facter.rb

```diff
@@ -21,6 +21,13 @@ def save(facts)
   # Lookup a host's facts up in Facter.
   def find(request)
     Facter.reset
+
+    # Add the puppetversion fact; this is done before generating the hash so it is
+    # accessible to custom facts.
+    Facter.add(:puppetversion) do
+      setcode { Puppet.version.to_s }
+    end
+
```

Comment:
The only route that seems to make sense is to separate what fact/value pairs we want to add from how they're added, then do it before to_hash for factor facts and after we create the hash for networking facts.